src/main/scala/com/fortysevendeg/android/sqlite/statement/StatementArguments.scala

```diff
@@ -0,0 +1,170 @@
+    Option(value) match {
+      case Some(v: String) => v
+      case Some(v) => v.toString
+      case None => javaNull
+    }
+  }
+
+  override def toArray: Array[AnyRef] = toArray(argumentsList.head, maxIndex)
+
+  // In android only byte[], String, Long and Double are supported in bindArgs.
+
+  def setObjectArgument(position: Int, arg: Any): Unit = withValidation(position, arg) {
+    case a: scala.Array[Byte] => setArgument(position, a)
+    case a: Byte => setArgument(position, a)
+    case a: Boolean => setArgument(position, a)
+    case a: Double => setArgument(position, a)
+    case a: Short => setArgument(position, a)
+    case a: Float => setArgument(position, a)
+    case a: Long => setArgument(position, a)
+    case a: String => setArgument(position, a)
+    case a: java.sql.Date => setArgument(position, a)
+    case a: java.sql.Time => setArgument(position, a)
+    case a: java.sql.Timestamp => setArgument(position, a)
+    case `javaNull` => setNullArgument(position)
+    case _ => throw new SQLException(classNotSupportedErrorMessage(arg.getClass.getName))
+  }
+
+  def setArgument(position: Int, arg: scala.Array[Byte]): Unit =
+    withValidation(position, arg)(addToHead(position, _))
+
+  def setArgument(position: Int, arg: Double): Unit =
+    withValidation(position, arg) { d =>
+      addToHead(position, java.lang.Double.valueOf(d))
+    }
+
+  def setArgument(position: Int, arg: Long): Unit =
+    withValidation(position, arg) { l =>
+      addToHead(position, java.lang.Long.valueOf(l))
+    }
+
+  def setArgument(position: Int, arg: String): Unit =
+    withValidation(position, arg)(addToHead(position, _))
+
+  def setArgument(position: Int, arg: Byte): Unit =
+    withValidation(position, arg) { a =>
+      setArgument(position, a.toLong)
+    }
+
+  def setArgument(position: Int, arg: Boolean): Unit =
+    withValidation(position, arg) { a =>
+      setArgument(position, a.toLong)
+    }
+
+  def setArgument(position: Int, arg: Short): Unit =
+    withValidation(position, arg) { a =>
+      setArgument(position, a.toLong)
+    }
+
+  def setArgument(position: Int, arg: Float): Unit =
+    withValidation(position, arg) { a =>
+      setArgument(position, a.toDouble)
+    }
+
+  def setArgument(position: Int, arg: java.sql.Date): Unit =
+    withValidation(position, arg) { a =>
+      setArgument(position, a.toLong)
+    }
+
+  def setArgument(position: Int, arg: java.sql.Time): Unit =
+    withValidation(position, arg) { a =>
+      setArgument(position, a.toLong)
+    }
+
+  def setArgument(position: Int, arg: java.sql.Timestamp): Unit =
+    withValidation(position, arg) { a =>
+      setArgument(position, a.toLong)
+    }
+
+  private[this] def addToHead(position: Int, arg: AnyRef) = {
+    argumentsList.head += (position -> arg)
+    if (position > maxIndex) maxIndex = position
+  }
+
+  private[this] def setNullArgument(position: Int): Unit = {
+    argumentsList.head.remove(position)
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Usage of head/last on collections.](https://www.codacy.com/app/47deg/scala-sqlite-droid/file/821582812/issues/source?bid=2442814&fileBranchId=2752104#l124)